src/components/inventory/inventory.modal.component.js

```diff
@@ -21,123 +22,189 @@ class InventoryDetailsModal extends Component {
       isLoading: true,
       modalOpen: true,
       item: this.props.selectedItem.item,
-      activeIndex: this.props.selectedItem.index 
+      activeIndex: this.props.selectedItem.index,
     };
-    this.hanldleTagSelected = this.hanldleTagSelected.bind(this)
+    this.hanldleTagSelected = this.hanldleTagSelected.bind(this);
   }
 
-  handleClose = e => {
+  UNSAFE_componentWillMount() {
+    this.getGatewaysLocations();
+  };
+
+  UNSAFE_componentWillReceiveProps(newProps){
+    this.setState({
+      item: newProps.selectedItem.item,
+      activeIndex: newProps.selectedItem.index,
+    }, this.getGatewaysLocations);
+  }
+
+  getGatewaysLocations() {
+    const paramsId = {
+      type: this.props.selectedItem.item.type.toLowerCase(),
```

Comment:
If it is possible that type comes with undefined or null, you will have a problem here.... may be you can use something like this: 
```suggestion
      type: _.lowerCase(this.props.selectedItem.item.type),
```